aptos-move/framework/move-stdlib/sources/vector.move

```diff
@@ -228,6 +268,94 @@ module std::vector {
         result
     }
 
+    /// Partition, sorts all elements for which pred is true to the front.
+    /// Preserves the relative order of the elements for which pred is true,
+    /// BUT NOT for the elements for which pred is false.
+    public inline fun partition<Element>(
+        v: &mut vector<Element>,
+        pred: |&Element|bool
+    ): u64 {
+        let i = 0;
+        let len = length(v);
+        while (i < len) {
+            if (!pred(borrow(v, i))) break;
+            i = i + 1;
+        };
+        let p = i;
+        i = i + 1;
+        while (i < len) {
+            if (pred(borrow(v, i))) {
+                swap(v, p, i);
+                p = p + 1;
+            };
+            i = i + 1;
+        };
+        p
+    }
+
+    fun test_stable_partition() {
+        let v = vector[1, 2, 3, 4, 5];
+        assert!(stable_partition(&mut v, |n| *n % 2 == 0) == 2, 0);
+        assert!(&v == &vector[2, 4, 1, 3, 5], 1);
+    }
+
+    /// rotate(&mut [1, 2, 3, 4, 5], 2) -> [3, 4, 5, 1, 2] in place, returns the split point
+    /// ie. 3 in the example above
+    public fun rotate<Element>(
+        v: &mut vector<Element>,
+        rot: u64
+    ): u64 {
+        let len = length(v);
```

Comment:
nit: Why not inline length(v) on line 309?

Reply:
not allowed, first parameter v is mut and length takes another borrow